cron/rss_reader.py

```diff
@@ -10,11 +10,12 @@ def __init__(self, feed_url):
     def read_feed(self, newer_than):
         feed = feedparser.parse(self.feed_url)
 
+        print(len(feed.entries))
+
         for idx, entry in enumerate(feed.entries):
-            published_ts = datetime.fromisoformat(entry["published"].replace("Z", ""))
+            published_ts = datetime.strptime(entry["published"], "%Y-%m-%dT%H:%M:%SZ")
+
+            if published_ts < newer_than:
+                return
 
-            if idx == 0:
-                if published_ts < newer_than:
-                    return
-            else:
-                yield entry
+            yield entry
```

Comment:
I think this logic is correct, but it's a little hard to reason about because it still relies on the feed being sorted a certain way. I think if we dropped the early return and just always checked all entries, that would make it not just correct, but *obviously* correct. Like:

```python
for ...:
    published_ts = ...
    if published_ts >= newer_than:
        yield entry
# that's all, no return anywhere
```

Reply:
I agree on it